### ISSUE TITLE
Make APIs type safe according to AAS meta model spec

### ISSUE THREAD

**Issue by atextor** [eclipse-aas4j/aas4j#197]
Many properties in the AAS meta model are optional. However, in the AAS4J API, corresponding methods return `null` when a certain value is not present in a loaded Environment. One (arbitrarily chosen) example: [`AdministrativeInformation#getVersion()`](https://github.com/eclipse-aas4j/aas4j/blob/6b2fab7dae2cbb0c2da2436524c5f268258e99ec/model/src/main/java/org/eclipse/digitaltwin/aas4j/v3/model/AdministrativeInformation.java#L40) returns the value of an optional field, but neither the API nor the JavaDoc indicate this. The only way to find out is by opening the [link](https://aas-core-works.github.io/aas-core-meta/v3/AdministrativeInformation.html#version) and look at the specification.

IMO adding this information in the JavaDoc (i.e., "@&#8203;returns Returns the String for the property version _or null_") would only be a weak workaround; I'd actually like to have such methods return `java.util.Optional<String>`. This is the only way to work robustly with the API.

**Comment by sirea-07** [comment]
The problem with `java.util.Optional` is that it's [not intended as a "maybe type"](https://stackoverflow.com/a/26328555/19399702), which obviously could deliberately be ignored, but more importantly it doesn't support `null` as a value. If there is a difference between `null` as a value and the property not being present, `Optional` cannot be used. Now I don't know if this is relevant for any property in AAS, but as an example due to this issue, OpenAPITools has added the type [JsonNullable](https://github.com/OpenAPITools/jackson-databind-nullable).

**Comment by atextor** [comment]
Yes, I'm aware of how `Optional` was originally intended. From the post you linked, citing Brian Goetz:

> Our intention was to provide a limited mechanism for library method return types where there needed to be a clear way to represent "no result", and using null for such was overwhelmingly likely to cause errors. 

This is exactly what we have here, and what I'm requesting.

I do know that in some cases distinguishing between present/not present/null is necessary (in particular for PUT/PATCH requests in REST APIs) and I also know JsonNullable. However, I don't think this is relevant here. The AAS meta model clearly states that some fields are optional, a value of `null` to indicate anything specifically, however, is _not_ mentioned anywhere. Therefore `Optional` both can be used, and is used in the way it was intended.

**Comment by sirea-07** [comment]
Although I have a different interpretation of that quote, it doesn't really matter. As long as there is no need to differentiate between `null` and a value not being present, `Optional` is fine to use imo as well and I support your suggested change.

**Comment by mjacoby** [comment]
What exactly is the benefit of having optional properties return `Optional<>`? A getter that returns an `Optional<>` can still be null (if not explicitly enforced, e.g. in every setter method), i.e. you still have to check if the value is null before accessing it which means at the end of the day you need to check if the return value is null and if not again check `Optional.isPresent()`. 
In this case, what would be the semantics of each of those cases and wouldn't they in the end not have the same meaning?

It maybe is helpful that the AAS metamodel classes in AAS4J are designed in such a way that it is possible to create instances that are not valid, e.g. by not setting the idShort where it is actually required. We made the design decision that we think it is better to allow the creation of invalid instances and than add a validator component that will check if a given instance if compliant to the standard. Unfortunately, this validator is currently not available as we did not yet finish updating it to the metamodel v3.0. 

Generally speaking, there are multiple things the AAS metamodel classes can be used for, e.g. as result format from loading AAS model files, to create AAS models from code or as payload for API operations. The only scenario I really think using `Optional<>` could be beneficial is when using the classes as payload for PATCH commands because in this case there is a strong semantic difference between a property being not present (meaning it should not be changed) or it being null (meaning it should be set to null, i.e. deleted/removed).
If you use AAS4J in this case to parse the payload, this information is lost, i.e. the metamodel classes are not helpful in this case.

My approach upon implementing PATCH using AAS4J was therefore not to parse the payload to the AAS4J classes but to treat it as a generic JSON Merge request and use libraries that can execute any JSON Merge request on any object instance.

Coming back to the potential implications of using `Optional<>` and assuming we would enforce that properties of this type could never actually be null but only `Optional.empty`, how would you suggest to handle collections that can be empty, e.g. Submodel.submodelElement or SubmodlElementCollection.value and so on? As they have a cardinality of `0..*` I would consider them optional and therefore would expect the API to return something like `Optional<Collection<>>` which seems cumbersome. 

For clarification, I do not categorically oppose the use of `Optional<>` but to make an informed decision we need a detailed list of pros and cons. If you think we should use `Optional<>` please provide your arguments in a comprehensible way, e.g. why it is not suitable to check if [AdministrativeInformation#getVersion()](https://github.com/eclipse-aas4j/aas4j/blob/6b2fab7dae2cbb0c2da2436524c5f268258e99ec/model/src/main/java/org/eclipse/digitaltwin/aas4j/v3/model/AdministrativeInformation.java#L40) returns `null` instead of `Optional.empty`.

**Comment by mjacoby** [comment]
After discussing this in the team multiple times since creation of this ticket we do not see the benefits of using `Optional<>` and therefore will keep the current behavior which returns `null` instead.
